Description: Make key-above-tab when using application switcher more consistent
 If you alt-tab, have the switcher open on an application displaying multiple
 windows, you have multiple cases there:
 - if you press down, the window thumbnails are previewed, and first element
   is selected
 - if you press key-above-tab, the window thumbnails are previewed, however
   the second element is directly selected.
 This isn't consistent with what the user can expect. This patch allow that
 the first key-above-tab press select the first window in the item if the
 thumbnail view wasn't selected.
Origin: ubuntu
Bug-Ubuntu: https://launchpad.net/bugs/1716872
Forwarded: https://bugzilla.gnome.org/show_bug.cgi?id=786009
Index: gnome-shell-3.24.3/js/ui/altTab.js
===================================================================
--- gnome-shell-3.24.3.orig/js/ui/altTab.js
+++ gnome-shell-3.24.3/js/ui/altTab.js
@@ -159,7 +159,11 @@ const AppSwitcherPopup = new Lang.Class(
 
     _keyPressHandler: function(keysym, action) {
         if (action == Meta.KeyBindingAction.SWITCH_GROUP) {
-            this._select(this._selectedIndex, this._nextWindow());
+            if (Desktop.is("ubuntu") && !this._thumbnailsFocused) {
+                this._select(this._selectedIndex, 0);
+            } else {
+                this._select(this._selectedIndex, this._nextWindow());
+            }
         } else if (action == Meta.KeyBindingAction.SWITCH_GROUP_BACKWARD) {
             this._select(this._selectedIndex, this._previousWindow());
         } else if (action == Meta.KeyBindingAction.SWITCH_APPLICATIONS) {
